src/software/ai/hl/stp/tactic/tactic.cpp

```diff
@@ -107,3 +109,37 @@ std::set<RobotCapability> &Tactic::mutableRobotCapabilityRequirements()
 {
     return capability_reqs;
 }
+
+std::unique_ptr<Intent> Tactic::get(const Robot &robot, const World &world)
+{
+    // TODO (#1888): remove these updates
+    updateWorldParams(world);
+    updateRobot(robot);
+
+    updateIntent(TacticUpdate{.robot      = robot,
+                              .world      = world,
+                              .set_intent = [this](std::unique_ptr<Intent> new_intent) {
+                                  intent = std::move(new_intent);
+                              }});
+    if (intent)
+    {
+        return std::move(intent);
+    }
+    else
+    {
+        LOG(WARNING) << "No intent set for this tactic" << std::endl;
```

Comment:
This is in `tactic.cpp` we won't know which tactic this is, maybe worth adding `TYPENAME(*this)` here (although not sure if that will even work)?

Reply:
done